packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -122,22 +122,36 @@ abstract class LambdaAuthorizer extends Authorizer implements IAuthorizer {
    */
   protected setupPermissions() {
     if (!this.role) {
-      this.handler.addPermission(`${Names.uniqueId(this)}:Permissions`, {
-        principal: new iam.ServicePrincipal('apigateway.amazonaws.com'),
-        sourceArn: this.authorizerArn,
-      });
-    } else if (this.role instanceof iam.Role) { // i.e. not imported
-      this.role.attachInlinePolicy(new iam.Policy(this, 'authorizerInvokePolicy', {
-        statements: [
-          new iam.PolicyStatement({
-            resources: this.handler.resourceArnsForGrantInvoke,
-            actions: ['lambda:InvokeFunction'],
-          }),
-        ],
-      }));
+      this.addDefaultPermisionRole();
+    } else if (this.role instanceof iam.Role) {
```

Comment:
`instanceof` is not actually a reliable test of type. While you're refining this, could you update this? See https://github.com/aws/aws-cdk/pull/20564 for how we started changing this elsewhere.

Reply:
@TheRealAmazonKendra 

Thank you for teaching me!

I didn't know InstanceOf worked, so I learned a lot!

Keep me posted if you need anything else!

Very informative and fun!

I tried it.

Could you review please?

https://github.com/aws/aws-cdk/pull/24340/files#diff-6da774d31de115da8b25000e51fe2dd4690c47f94b9429006cce70f88862149dR304-R306